### ISSUE TITLE
Pure Fans using the Same Controller

### ISSUE THREAD

**Issue by natsukirei** [chestm007/linux_thermaltake_riing#33]
Hey Team,

I was just wondering if this program would work for the Pure Plus fans, they use the same controller as the Riing packs and Fan Packs, with the same type of Proprietary plugs.

I miss linux but ive been forced to use windows due to lack of fan control over my fans, i dont actually care about the RGB, i mostly want speed to temp control over the system fans.

https://www.thermaltake.com.au/C_00003232.htm

Also i would like to add that i would love to see this fan control made into app packages such as snap, appimages and flatpaks that would be amazing to add automated distro support

**Comment by chestm007** [comment]
So i think it should work - try setting the controller config section as

```
controllers:
  - unit: 1
    type: g3
    devices:
      1: Riing Plus
      2: Riing Plus
      3: Riing Plus
```

**Comment by snuffysasa** [comment]
Can confirm it does work

**Comment by natsukirei** [comment]
> 
> 
> Can confirm it does work

snuffy, could you share your configs and scripts possibly? im new to this, i wouldnt even have a clue how to set it up to control fans on reboot etc

kinda wish this program had a gui control center

**Comment by snuffysasa** [comment]
Yes I can share the configs and scripts and help you get it setup.

I also plan to build a web gui for myself to use to control the fans, which I can also share.

**Comment by natsukirei** [comment]
> 
> 
> Yes I can share the configs and scripts and help you get it setup.
> 
> I also plan to build a web gui for myself to use to control the fans, which I can also share.

that would be amazing, this is a very unique program that was made for a unique setup, thermaltake is quite popular so i am sure this would see some use.

i think this program deserves a good GUI similar to GreenWithEnvy for nvidia GPU control, which could sit in the taskbar in most Graphical environments such as Gnome and boot with linux.

If you could achieve that, much praise to you

**Comment by snuffysasa** [comment]
Should be able to work on this some over this coming weekend.  I keep getting lots of ideas of features, though I want to stay focused and just start with getting baseline core / minimum features.  What do you see as features in a minimum viable product? So I can start there.

I looked at GreenWithEnvy.... I could so something kind of like that but start with less feature rich. As far as making a draggable line graph to select rpm to temp ramps... I'll need to look to see if there is some JS tool that would work well for that. (probably is), but probably start with something more simple.

As far as booting up with linux.... I'll probably just make instruction steps in the readme to make it the program launch on startup... though it will be different for each linux distro. Are you using ubuntu?

And as far is the GUI... I'll probably do web based because I could do that quicker and will be compatible with all systems and could be accessed from other computers and mobile (if you wanted to allow that)...  

so basically, some script will be setup to launch server when linux boots up, and then from there you'd just open up a web browser and type in localhost:3000 (or something) and then it loads a simple web app where you can see the config parameters and adjust them

**Comment by natsukirei** [comment]
> 
> 
> Should be able to work on this some over this coming weekend. I keep getting lots of ideas of features, though I want to stay focused and just start with getting baseline core / minimum features. What do you see as features in a minimum viable product? So I can start there.
> 
> (Spent some time on Sunday making some custom light shows.... but that's all just eye candy)
> 
> I looked at GreenWithEnvy.... I could so something kind of like that but start with less feature rich. As far as making a draggable line graph to select rpm to temp ramps... I'll need to look to see if there is some JS tool that would work well for that. (probably is), but probably start with something more simple.
> 
> As far as booting up with linux.... I'll probably just make instruction steps in the readme to make it the program launch on startup... though it will be different for each linux distro. Are you using ubuntu?
> 
> And as far is the GUI... I'll probably do web based because I could do that quicker and will be compatible with all systems and could be accessed from other computers and mobile (if you wanted to allow that)...
> 
> so basically, some script will be setup to launch server when linux boots up, and then from there you'd just open up a web browser and type in localhost:3000 (or something) and then it loads a simple web app where you can see the config parameters and adjust them

That all sounds fantastic.

In answer to your questions, i usually use manjaro and Solus, and i dont think it needs to be "feature rich" but, being able to adjust fan speed would be step 1, along with RGB later on beyond a basic color maybe.

I noticed in the software, i usually go between silent or performance settings for each fan, silent seems to be around 500-650 rpm, sometimes 800 rpm if it hits above 60C which is rare, i rarely go above 42C, and performance is between 700-1000 on average RPM.

So maybe having a silent and performance option , which sets the fans to 550 RPM/600 RPM, and 800-1000 RPM, which would be happy mediums and keep things simple.

**Comment by snuffysasa** [comment]
Just letting you know I haven't forgotten about this project, and I'll have something basic to send to you soon.

**Comment by snuffysasa** [comment]
@natsukirei 
I got something really basic working. Detects what Thermaltake Controllers are connected and how many fans and serves a basic web gui that lets you view and change speeds and set a temperature based speed.  Though, I'm not sure if the temp sensors will work correctly on other computers yet.

See this test release
https://gitlab.com/mattsasa/nodejsttrgb/-/releases/test-release

And I think we should move this discussion to [here](https://gitlab.com/mattsasa/nodejsttrgb/-/issues/1)
